tools/docker/run.sh

```diff
@@ -107,7 +110,7 @@ main() {
                 -e INSTALL_DIR=${installdir}
                 --privileged
                 --network ${NETWORK}
-                --expose ${PORT}
+                --expose ${PORT} ${PUBLISH}
```

Comment:
Any reason why not to always publish all ports?
What is the difference between exposing port and publishing it?

Reply:
You don't always want to publish the ports to the host.
Sometimes you only want to expose certain ports for container-to-container communication.

From the official docker documentation:

> The EXPOSE instruction defines the initial incoming ports that provide services. These ports are available to processes inside the container. An operator can use the --expose option to add to the exposed ports.
> 
> To expose a container’s internal port, an operator can start the container with the -P or -p flag. The exposed port is accessible on the host and the ports are available to any client that can reach the host.